src/render/posts/does-tdd-work.html.md

```diff
@@ -0,0 +1,228 @@
+supportive that the TDD approach yields code with superior
+external code quality. However, the validity of the results must be
+considered within the context of the limitations discussed in external
+validity section.
+
+There were also some interesting findings around the lack of testing done by the control group. Not practicing TDD appeared to cause a leaning towards a lack of testing in general. 
+
+This study had many limitations but it was still a good effort to gather some real data on how TDD might work in industry. In terms of proof that TDD works or not, this study is far from that. 
+
+An interesting paper to look at is [Janzen, D. S., (2006). An Empirical Evaluation of the Impact of
+Test-Driven Development on Software Quality](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.94.9412&rep=rep1&type=pdf) which looks at other studies and does analysis on data gathered by them. This is a long paper and I wont repeat much of it here but I will highlight some interesting data and points from it. The data is interesting but it is worth keeping in mind the data is from the early 2000's with relatively small sample sizes. 
+
+Below are tables with a summaries of findings from various papers referenced in [Janzen, D. S., (2006)](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.94.9412&rep=rep1&type=pdf)
+
+Type (CE) is a controlled experiment and (CS) is a case study.
+
+#### In Industry
+|Study|Type|No.of Companies|No. of Programmers|Quality Effects|Productivity Effects|
+|-----|----|---------------|------------------|---------------|--------------------|
+|[George](https://pdfs.semanticscholar.org/616d/3f7e831c725b51220a34fbee3ca6ac1d711c.pdf)|CE|3|24|TDD passed 18% more tests|TDD took 16% longer|
+|[Maximilien](http://faculty.salisbury.edu/~xswang/Research/Papers/SERelated/TDD/p564-maximilien.pdf)|CS|1|9|50% reduction in defect density|minimal impact|
+|[Williams](http://ieeexplore.ieee.org/document/1251029/)|CS|1|9|40% reduction in defect density|no change|
+
+#### In Academia
+|Study|Type|No. of Programmers|Quality Effects|Productivity Effects|
+|-----|----|------------------|---------------|--------------------|
+|[Edwards](http://www.cs.tufts.edu/~nr/cs257/archive/stephen-edwards/automated-feedback.pdf)|CE|59|54% fewer defects|n/a|
+|[Kaufmann](http://dl.acm.org/citation.cfm?id=949421)|CE|8|improved information flow|50% improvement|
+|[Muller](http://ieeexplore.ieee.org/document/1049202/)|CE|19|no change, but better reuse|no change|
+|[Pancur](http://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=1248175)|CE|38|no change|no change|
+|[Erdogmus](http://ieeexplore.ieee.org/document/1423994/)|CE|35|no change|improved productivity|
+
+The data here suggest mostly positive or neutral results from TDD. There is one exception (the previously mentioned George paper) where TDD took 16% longer than the control group but it is noted the control group wrote lest tests in that study.
```

Comment:
"wrote less tests", though actually I think "wrote fewer tests" would be more correct.